### PR TITLE
CC-1223: Refactor streams extension tests to use RESP framework

### PR DESCRIPTION
**Pull request by ryan-gang**
This pull request refactors the streams extension tests to use the RESP framework for the Redis client connection. It modifies the test cases to use the MultiCommandTestCase and resp_assertions for assertions. The changes improve the efficiency and readability of the code.

### CODE REVIEW

File: internal/test_streams_xadd_validate_id.go [hunk 1]
```diff
@@ -13,25 +16,33 @@ func testStreamsXaddValidateID(stageHarness *test_case_harness.TestCaseHarness)
...
+	randomValues := random.RandomWords(10)
+
+	multiCommandTestCase := test_cases.MultiCommandTestCase{
+		Commands: [][]string{
+			{"XADD", randomKey, "1-1", randomValues[0], randomValues[1]},
+			{"XADD", randomKey, "1-2", randomValues[2], randomValues[3]},
+			{"XADD", randomKey, "1-2", randomValues[4], randomValues[5]},
+			{"XADD", randomKey, "0-3", randomValues[6], randomValues[7]},
+			{"XADD", randomKey, "0-0", randomValues[8], randomValues[9]},
+		},
```
Comment (ryan-gang) [inline internal/test_streams_xadd_validate_id.go:37]: `MultiCommandTestCase` is such a nice abstraction. 
Makes the tests much more readable & concise.

Comment (rohitpaulk) [reply internal/test_streams_xadd_validate_id.go:37]: Agreed! Might be even more readable if `Assertions` & `Commands` appeared alongside each other though, slight hard to follow what the expected output for each command is with this structure.

**Review by rohitpaulk — APPROVED** [review]
